src/ProjectService.Business/Commands/Project/CreateProjectCommand.cs

```diff
@@ -34,36 +31,75 @@ public class CreateProjectCommand : ICreateProjectCommand
         private readonly ICreateProjectValidator _validator;
         private readonly ILogger<CreateProjectCommand> _logger;
-        private readonly IProjectInfoMapper _projectInfoMapper;
         private readonly IHttpContextAccessor _httpContextAccessor;
-        private readonly IRequestClient<IGetDepartmentRequest> _rcGetDepartment;
         private readonly IRequestClient<ICreateWorkspaceRequest> _rcCreateWorkspace;
+        private readonly IRequestClient<ICheckUsersExistence> _rcCheckUsersExistence;
         private readonly IRequestClient<ICreateWorkTimeRequest> _rcCreateWorkTime;
+        private readonly IRequestClient<ICheckDepartmentsExistence> _rcCheckDepartmentsExistence;
 
-        private IGetDepartmentResponse GetDepartment(Guid departmentId, List<string> errors)
+        private List<Guid> CheckDepartmentExistence(Guid? departmentId, List<string> errors)
         {
-            string errorMessage = "Cannot add project. Please try again later.";
+            if (!departmentId.HasValue)
+            {
+                return null;
+            }
+
+            string errorMessage = "Failed to check the existing department.";
+            string logMessage = "Department with id: {id} not found.";
 
             try
             {
-                var response = _rcGetDepartment.GetResponse<IOperationResult<IGetDepartmentResponse>>(
-                    IGetDepartmentRequest.CreateObj(null, departmentId), timeout: TimeSpan.FromSeconds(2)).Result;
-
+                var response = _rcCheckDepartmentsExistence.GetResponse<IOperationResult<ICheckDepartmentsExistence>>(
+                    ICheckDepartmentsExistence.CreateObj(new List<Guid> { departmentId.Value })).Result;
                 if (response.Message.IsSuccess)
                 {
-                    return response.Message.Body;
+                    if (!response.Message.Body.DepartmentIds.Any())
+                    {
+                        errors.Add($"Department Id: {departmentId} does not exist");
+                    }
+                    return response.Message.Body.DepartmentIds;
                 }
 
-                _logger.LogWarning($"Can not find department with this id '{departmentId}': " +
+                _logger.LogWarning("Can not find department with this Id: {departmentId}: " +
                     $"{Environment.NewLine}{string.Join('\n', response.Message.Errors)}");
             }
             catch (Exception exc)
             {
-                _logger.LogError(exc, errorMessage);
+                _logger.LogError(exc, logMessage);
+            }
 
-                errors.Add(errorMessage);
+            errors.Add(errorMessage);
+            return null;
+        }
+
+        private List<Guid> CheckUserExistence(List<Guid> userIds, List<string> errors)
+        {
+            if (!userIds.Any())
+            {
+                return userIds;
             }
 
+            string errorMessage = "Failed to check the existing users.";
```

Comment:
errorMessage nowhere used except in catch. Do not set errorMsg to a variable - set it to args. logMessage also